Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -208,6 +287,77 @@ void DiscusMultipleScatteringCorrection::getXMinMax(const Mantid::API::MatrixWor
+  convertWsBothAxesToPoints(m_SQWS);
+  // if S(Q,w) has been supplied ensure Q is along the x axis of each spectrum (so same as S(Q))
+  if (m_SQWS->getAxis(1)->unit()->unitID() == "MomentumTransfer") {
+    auto transposeAlgorithm = this->createChildAlgorithm("Transpose");
+    transposeAlgorithm->initialize();
+    transposeAlgorithm->setProperty("InputWorkspace", m_SQWS);
+    transposeAlgorithm->setProperty("OutputWorkspace", "_");
+    transposeAlgorithm->execute();
+    m_SQWS = transposeAlgorithm->getProperty("OutputWorkspace");
+  } else if (m_SQWS->getAxis(1)->isSpectra()) {
+    // for elastic set w=0 on the spectrum axis to align code with inelastic
+    auto newAxis = std::make_unique<NumericAxis>(std::vector<double>{0.});
+    newAxis->setUnit("DeltaE");
+    m_SQWS->replaceAxis(1, std::move(newAxis));
+  }
+
+  // generate log of the structure factor to support gaussian interpolation
+  m_logSQ = m_SQWS->clone();
+  for (size_t i = 0; i < m_logSQ->getNumberHistograms(); i++) {
+    auto &ySQ = m_logSQ->mutableY(i);
+
+    std::transform(ySQ.begin(), ySQ.end(), ySQ.begin(), [](double d) -> double {
+      const double exp_that_gives_close_to_zero = -20.0;
+      if (d == 0.)
+        return exp_that_gives_close_to_zero;
+      else
+        return std::log(d);
+    });
+  }
+}
+
+/**
+ * Convert x axis of a workspace to points if it's bin edges. If the spectrum axis is a numeric axis and also bin edges
+ * convert that to points as well
+ * @param ws The workspace that will potentially be converted
+ */
+void DiscusMultipleScatteringCorrection::convertWsBothAxesToPoints(MatrixWorkspace_sptr &ws) {
+  if (ws->isHistogramData()) {
+    if (!m_importanceSampling) {
+      auto pointDataAlgorithm = this->createChildAlgorithm("ConvertToPointData");
+      pointDataAlgorithm->initialize();
+      pointDataAlgorithm->setProperty("InputWorkspace", ws);
+      pointDataAlgorithm->setProperty("OutputWorkspace", "_");
+      pointDataAlgorithm->execute();
+      ws = pointDataAlgorithm->getProperty("OutputWorkspace");
+    } else {
+      // flat interpolation is later used on S(Q) so convert to points by assigning Y value to LH bin edge
+      MatrixWorkspace_sptr SQWSPoints =
+          API::WorkspaceFactory::Instance().create(ws, ws->getNumberHistograms(), ws->blocksize(), ws->blocksize());
+      SQWSPoints->setSharedY(0, ws->sharedY(0));
+      SQWSPoints->setSharedE(0, ws->sharedE(0));
+      std::vector<double> newX = ws->histogram(0).dataX();
+      newX.pop_back();
+      SQWSPoints->setSharedX(0, HistogramData::Points(newX).cowData());
+      ws = SQWSPoints;
+    }
+  }
+  if (dynamic_cast<BinEdgeAxis *>(ws->getAxis(1)) != nullptr) {
+    auto edges = dynamic_cast<NumericAxis *>(ws->getAxis(1))->getValues();
+    std::vector<double> centres(edges.size() - 1);
+    for (size_t i = 0; i < centres.size(); ++i) {
+      centres[i] = (0.5 * (edges[i] + edges[i + 1]));
```

Comment:
I'm pretty sure there's handling from edges -> centres in Mantid:

Calling `Histogram::Points` should avoid having to check and convert: https://docs.mantidproject.org/nightly/concepts/HistogramData.html#points